packages/react-vapor/src/components/validation/hoc/tests/WithDirtySingleSelectHOC.spec.tsx

```diff
@@ -0,0 +1,77 @@
+import { ShallowWrapper } from 'enzyme';
+import {mountWithStore, shallowWithStore} from 'enzyme-redux';
+import * as React from 'react';
+import { act } from 'react-dom/test-utils';
+import { Store } from 'redux';
+import { ValidationActions, ValidationTypes } from '../..';
+import { IReactVaporState } from '../../../../Entry';
+import {composeMockStore, getStoreMock, withSelectedValues} from '../../../../utils/tests/TestUtils';
+import { ISingleSelectOwnProps, SingleSelectConnected } from '../../../select/SingleSelectConnected';
+import { withDirtySingleSelectHOC } from '../WithDirtySingleSelectHOC';
+
+
+describe('SingleSelectWithDirty', () => {
+    const SingleSelectWithHOC = withDirtySingleSelectHOC(SingleSelectConnected);
+    let store: ReturnType<typeof getStoreMock>;
+
+    const ONE_VALUE = '🐟';
+    const ANOTHER_VALUE = '🐠';
+
+    const defaultProps: ISingleSelectOwnProps = {
+        id: 'SOME_ID',
+        items: [],
+    };
+
+    beforeEach(() => {
+        store = getStoreMock({
+            validation: {},
+        });
+    });
+
+    afterEach(() => {
+        store.clearActions();
+    });
+
+    const mountSingleSelectWithHOC = (defaultProps: ISingleSelectOwnProps, store: Store<IReactVaporState>, initialValue?: string) => {
```

Comment:
You certain you need to mount it? :thinking: I think you could simply mock `React.useEffect` to call it when mounting, and it should be good enough

Reply:
I see that the other dirty are tested like that, so let's keep it this way